Sources/PluginViewController/Configurations/SectionsConfiguration.swift

```diff
@@ -1,55 +1,135 @@
 //
 //  Copyright © 2017 Nikita Ermolenko. All rights reserved.
+//  Copyright © 2019 Rosberry. All rights reserved.
 //
 
 import Foundation
 
 final class SectionsConfiguration: NSObject, Configuration {
 
-    let sections: [Section]
+    let plugins: [Plugin]
+
+    var sections: [NavigationPlugin] = []
+    var tableViewItems: [PluginItem] = []
+
     weak var tableView: UITableView?
+    weak var delegate: ConfigurationDelegate?
 
-    init(sections: [Section]) {
-        self.sections = sections
+    init(plugins: [Plugin]) {
+        self.plugins = plugins
     }
 
     func configure() {
-        sections.flatMap { section in
-            return section.plugins
-        }.forEach { plugin in
-            self.tableView?.register(plugin.nib, forCellReuseIdentifier: plugin.cellIdentifier)
+        sections = self.sections(for: plugins)
+        tableViewItems = sections.map({ (section: NavigationPlugin) -> PluginItem in
+            return pluginItem(for: section)
+        })
+    }
+
+    // MARK: - Private
+
+    private func sections(for plugins: [Plugin]) -> [NavigationPlugin] {
+        var sections: [NavigationPlugin] = []
+        var sectionlessPlugins: [Plugin] = []
+        plugins.forEach { (plugin: Plugin) in
+            guard let section = plugin as? NavigationPlugin else {
+                sectionlessPlugins.append(plugin)
+                return
+            }
+
+            if sectionlessPlugins.isEmpty == false {
+                sections.append(SectionPlugin(plugins: sectionlessPlugins))
+                sectionlessPlugins.removeAll()
+            }
+
+            sections.append(section)
+        }
+
+        if sectionlessPlugins.isEmpty == false {
+            sections.append(SectionPlugin(plugins: sectionlessPlugins))
         }
+
+        return sections
+    }
+
+    private func register(_ cellClass: UITableViewCell.Type) {
+        tableView?.register(cellClass, forCellReuseIdentifier: cellClass.description())
+    }
+
+    private func pluginItem(for section: NavigationPlugin) -> PluginItem {
+        let isSingleSection = sections.count == 1
+        let children = pluginItems(for: section.plugins)
+
+        register(section.cellClass)
+        switch section.style {
+        case .plain:
+            return PluginItem(title: section.title.string, plugin: section, children: children)
+        case .nested:
+            return isSingleSection ?
+                PluginItem(title: nil, plugin: section, children: children) :
+                navigationItem(for: section)
+        }
+    }
+
+    private func pluginItems(for plugins: [Plugin]) -> [PluginItem] {
+        return plugins.map({ (plugin: Plugin) -> PluginItem in
```

Comment:
Is it possible to omit `(` ?
```suggestion
        return plugins.map { (plugin: Plugin) -> PluginItem in
```